keps/sig-node/3983-drop-in-configuration/README.md

```diff
@@ -0,0 +1,372 @@
+
+
+### Version Skew Strategy
+
+All behavior change is encapsulated within the Kubelet, so there is no version skew possible within core Kubernetes. It is possible third party tools may attempt to utilize the Kubelet’s drop-in directory before the Kubelet is upgraded to support it, which would cause silent failures. It is the responsibility of these third party tools to ensure the Kubelet is new enough to support this.
+
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [ ] Feature gate
+  - Feature gate name: KubeletDropInConfig
+  - Components depending on the feature gate: Kubelet
+
+Aside from the featuregate, admins will also have to explicitly enable the kubelet flag to enable this.
+
+
+###### Does enabling the feature change any default behavior?
+
+No, upgrading to a version of the Kubelet with this feature will not enable the Kubelet to be configured with the drop-in directory if no flag is specified.
+
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Roll back and remove the `--config-dir` flag from the kubelet's CLI.
+
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+This feature will be re-enabled via adding back the `--config-dir` flag to the CLI, as mentioned above.
+
+
+###### Are there any tests for feature enablement/disablement?
+
+A test will be added to assemble a single, functional kubelet configuration object from various individual drop-in config files.
+
+
+### Rollout, Upgrade and Rollback Planning
+
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The feature can cause the Kubelet to fail if the configuration in the drop-in directory is invalid. A rollback could fail if the original configuration also has an invalid configuration. This situation would cause workloads to not appear on that node. Neither of these cases are expected.
+
+
+###### What specific metrics should inform a rollback?
+
+The Kubelet not starting, which will cause nodes to be NotReady.
+
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Not specifically
+
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+
+### Monitoring Requirements
+
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Workloads do not directly consume this feature, it is for cluster admins during kubelet configuration.
```

Comment:
However, administrators can check the kubelet feature flag metric xxxx to see if it is enabled on a node.

You'll have to look up what that is called...

Reply:
I think you're referring to https://kubernetes.io/docs/reference/instrumentation/metrics/#list-of-alpha-kubernetes-metrics ->

```

kubernetes_feature_enabled | ALPHA | Gauge | This metric records the data about the stage and enablement of a k8s feature.

```

The flag itself is in alpha, does that affect anything?

Reply:
I don't think so. I think that is more for users to know the deprecation lifecycle of the metric. @logicalhan ?

Reply:
We're going to promote this soon.